packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -84,7 +84,8 @@ export async function generateBlogPosts(
 ) {
   const {include, routeBasePath} = options;
 
-  if (!fs.existsSync(blogDir)) {
+  const dirExist = await fs.pathExists(blogDir);
```

Comment:
Awaiting in a single statement is the same as synchronous. If there are no parallel async operations, it won't make a difference. But this makes it easier to write parallel code.

Reply:
this is async hooks on plugins so there is a high chance that there is another paralel async operation.

Reply:
Yes but within this async hook, the code is still synchronous and using `await` like this won't change that. Doing this doesn't let you interrupt the execution of one hook with another hook, the hook still runs to completion before the other starts.

Reply:
yep actually this one change doesnt really matter much.